src/lib/eina/_eina_fnmatch.c

```diff
@@ -0,0 +1,256 @@
+                         }
+                       else
+                         {
+                            strcpy_s(reg_pattern + i,
+                                     sizeof(reg_pattern + i),
+                                     "[^/]");
+                            i += 4;
+                         }
+                    }
+                  else if (pattern[j] == '*')  // PATHNAME 
+                    {
+                       strcpy_s(reg_pattern + i,
+                                sizeof(reg_pattern + i),
+                                "[^/]*");
+                       i += 5;
+                    }
+                  else
+                    {
+                       strcpy_s(reg_pattern + i,
+                                sizeof(reg_pattern + i),
+                                "[^/]");
+                       i += 4;
+                    }
+               }
+             else if (flags & FNM_PERIOD)
+               {
+                  if (j == 0)  // period at init
+                    {
+                       if (pattern[j] == '*')
+                         {
+                            strcpy_s(reg_pattern + i,
+                                     sizeof(reg_pattern + i),
+                                     "[\\.]*");
+                            i += 5;
+                         }
+                       else
+                         {
+                            strcpy_s(reg_pattern + i,
+                                     sizeof(reg_pattern + i),
+                                     "[\\.]");
+                            i += 4;
+                         }
+                    }
+                  else if (pattern[j] == '*')  // period at other places 
+                    {
+                       strcpy_s(reg_pattern + i,
+                                sizeof(reg_pattern + i),
+                                ".*");
+                       i += 2;
+                    }
+                  else
+                    reg_pattern[i++] = '.';
+               }
+             else if (pattern[j] == '*')  // NORMAL
+               {
+                  strcpy_s(reg_pattern + i,
+                           sizeof(reg_pattern + i),
+                           ".*");
+                  i += 2;
+               }
+             else
+               {
+                  strcpy_s(reg_pattern + i,
+                           sizeof(reg_pattern + i),
+                           ".");
+                  i++;
+               }
+          }
+        else if (pattern[j] == '.')
+          {
+             strcpy_s(reg_pattern + i, 
+                      sizeof(reg_pattern + i),
+                      "[\\.]");
+             i += 4;
+          }
+        else reg_pattern[i++] = pattern[j];  // OTHERS
+     }
+   strcpy_s(reg_pattern + i, 
+            sizeof(reg_pattern + i),
+            "$\0");
+
+   return reg_pattern;
+}
+
+int
+fnmatch(const char *pattern, const char *string, int flags)
+{
+   regex_t regex;
```

Comment:
I suggest moving this to below, above `if (regcomp(&regex...))`. It would point out better when is it initialized.